torch/_inductor/codegen/common.py

```diff
@@ -551,6 +552,30 @@ def deduce_output_dtype_by_name(
     return None
 
 
+def check_dtype(
+    buffer: IndentedBuffer, var: CSEVariableType, dtype: torch.dtype
+) -> None:
+    backend = get_current_backend()
+    if config.test_configs.runtime_triton_dtype_assert and backend == "triton":
+        buffer.writeline(f"tl.static_assert({var}.dtype == {triton_type(dtype)})")
```

Comment:
Curious why we insert `tl.static_assert` which is a compile-time check but name it `runtime_triton_dtype_assert` here?

Reply:
I'm not sure about this. It's ported from previous implementation.

https://github.com/pytorch/pytorch/commit/fd553b9817cb145e3cfaa52a08a831b9edf2fec7#diff-ecbd0bdf6ee8fc4eab27af1ec0b4369962dd340547451ac2f396b8f12f2f41bcR1810-R1822

Reply:
compile time is maybe more accurate.. debug config option anyway.. 🤷